src/common/transformations/src/transformations/common_optimizations/concat_reduce_fusion.cpp

```diff
@@ -0,0 +1,155 @@
+        return true;
+    };
+
+    auto m = std::make_shared<pattern::Matcher>(squeeze_pattern, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+ngraph::pass::ConcatReduceFusionWithoutFolding::ConcatReduceFusionWithoutFolding() {
+    MATCHER_SCOPE(ConcatReduceFusionWithoutFolding);
+
+    auto concat_pattern = ngraph::pattern::wrap_type<opset8::Concat>({pattern::any_input(), pattern::any_input()});
+    auto reduce_axes_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    auto reduce_pattern = ngraph::pattern::wrap_type<opset8::ReduceMin, opset8::ReduceMax>({concat_pattern, reduce_axes_pattern});
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        auto concat = as_type_ptr<opset8::Concat>(pattern_map.at(concat_pattern).get_node_shared_ptr());
+
+        ReduceType reduce_type = get_reduce_type(pattern_map.at(reduce_pattern).get_node_shared_ptr());
+        if (reduce_type == ReduceType::NONE) {
+            return false;
+        }
+
+        auto reduce = as_type_ptr<op::util::ArithmeticReductionKeepDims>(pattern_map.at(reduce_pattern).get_node_shared_ptr());
+
+        if (!reduce || !concat) {
+             return false;
+        }
+
+        const auto& reduction_axes = reduce->get_reduction_axes();
+        if (reduction_axes.size() != 1 || concat->get_axis() != static_cast<int64_t>(*reduction_axes.begin())) {
+            return false;
+        }
+
+        std::shared_ptr<ngraph::Node> result_node;
+        switch (reduce_type) {
+        case ReduceType::MAX:
+            result_node = register_new_node<opset8::Maximum>(concat->input_value(0), concat->input_value(1));
+            break;
+        case ReduceType::MIN:
+            result_node = register_new_node<opset8::Minimum>(concat->input_value(0), concat->input_value(1));
+            break;
+        default:
+            return false;
+        }
+
+        copy_runtime_info({concat, reduce}, result_node);
+
+        if (!reduce->get_keep_dims()) {
+            auto squeeze_axis_node = ngraph::opset8::Constant::create(ngraph::element::i64, {}, {*reduction_axes.begin()});
+            result_node = register_new_node<ngraph::opset8::Squeeze>(result_node, squeeze_axis_node);
+        }
+
+        result_node->set_friendly_name(reduce->get_friendly_name());
+
+        replace_node(reduce, result_node);
+
+        return true;
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(reduce_pattern, matcher_name);
+    register_matcher(m, callback);
+}
+
+bool ngraph::pass::ConcatReduceFusion::run_on_model(const std::shared_ptr<ngraph::Function>& f) {
+    RUN_ON_FUNCTION_SCOPE(ConcatReduceFusion);
+    ngraph::pass::Manager manager;
+    manager.register_pass<ConcatReduceFusionWithoutFolding>();
+    manager.register_pass<PullSqueezeThroughEltwise>();
+    manager.register_pass<ngraph::pass::NopElimination>();
```

Comment:
Can we combine this transformations in a single graph traversal? Also please use specific passes from NopElimination.

Reply:
Done, thanks!